lib/auth/auth.go

```diff
@@ -1173,15 +1183,16 @@ func (a *Server) GetMFADevicesWithToken(ctx context.Context, req *proto.GetMFADe
 	}, nil
 }
 
-// DeleteMFADeviceWithToken deletes a mfa device.
+// DeleteMFADeviceWithToken deletes a mfa device for user defined in token.
 func (a *Server) DeleteMFADeviceWithToken(ctx context.Context, req *proto.DeleteMFADeviceWithTokenRequest) error {
 	token, err := a.GetUserToken(ctx, req.GetTokenID())
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
 	// Restrict to certain kinds of user token and usage.
-	if token.GetSubKind() != UserTokenTypeRecoveryApproved || token.GetUsage() != types.UserTokenUsage_RECOVER_2FA {
+	not2FAApprovedTokenType := token.GetSubKind() != UserTokenTypeRecoveryApproved || token.GetUsage() != types.UserTokenUsage_RECOVER_2FA
+	if not2FAApprovedTokenType && token.GetSubKind() != UserTokenTypePrivilege {
```

Comment:
This conditional is getting a little bit hard to follow. I would either expand in the comments or even refactor it into a helper method.